models/contacts.js

```diff
@@ -1,19 +1,115 @@
-// const fs = require('fs/promises')
+import * as path from 'path';
+import { readFile, writeFile } from 'fs/promises';
+import { nanoid } from 'nanoid';
 
-const listContacts = async () => {}
+const contactsPath = path.resolve('./models', 'contacts.json');
 
-const getContactById = async (contactId) => {}
+const readContactsFile = async () => {
+  try {
+    const contactsJson = await readFile(contactsPath);
+    const contacts = JSON.parse(contactsJson);
+    return contacts;
+  } catch (err) {
+    console.error('Error reading contacts from file: ', err);
+    throw err;
+  }
+};
 
-const removeContact = async (contactId) => {}
+const writeContactsFile = async data => {
+  try {
+    await writeFile(contactsPath, JSON.stringify(data, null, 2));
+  } catch (err) {
+    console.error('Error writing contacts to file: ', err);
+    throw err;
+  }
+};
 
-const addContact = async (body) => {}
+export const listContacts = async () => {
+  try {
+    const contacts = await readContactsFile();
+    if (contacts) return contacts;
+    if (!contacts) console.log(`No contacts found`);
```

Comment:
Well, i think its enough if your try body will be only
```
try {
return await readContactsFile();
}
```
other checks are not necessary in my opinion. If you delete all contacts you should get empty array in return which is fine. Defenietly you shouldnt return false from this method in any case